src/acquisition/covidcast/file_archiver.py

```diff
@@ -0,0 +1,57 @@
+"""Moves files into various archival directories."""
+
+# standard library
+import gzip
+import os
+import shutil
+
+
+class FileArchiver:
+  """Archives files by moving and compressing."""
+
+  @staticmethod
+  def archive_file(
+      path_src,
+      path_dst,
+      filename,
+      compress,
+      gzip=gzip,
+      os=os,
+      shutil=shutil,
+      open_impl=open):
+    """Archive a file and return the path and `stat` of the destination file.
+
+    WARNING: This is a potentially destructive operation. See details below.
+
+    path_src: the directory which contains the file to be archived
+    path_dst: the directory into which the file should be moved
+    filename: the name of the file within `path_src`
+    compress: gzips the file if true, otherise moves the file unmodified
+
+    The destination directory will be created if necessary. If the destination
+    file already exists, it will be overwritten.
```

Comment:
do you want to perhaps at least log when this happens since it might indicate something unusual going on?

Reply:
done (and added new test)